### ISSUE TITLE
CSP smoke test

### ISSUE THREAD

**Issue by adamraine** [GoogleChrome/lighthouse#12378]
We should have a smoke test that tests Lighthouse on a page with a very strict CSP `default-src 'none'` to make sure a CSP doesn't block Lighthouse from running.

Some implementation notes:
- I think we should make a separate smoke category for csp tests. This doesn't fall neatly into any category.
- We should probably deliver the CSP through an HTTP header so we have access to the `sandbox` directive which was causing issues in https://github.com/GoogleChrome/lighthouse/issues/11925. Unfortunately this means we need to find a way to deliver the CSP header specifically for this smoke test.

Related:
https://github.com/GoogleChrome/lighthouse/issues/7159#issuecomment-744058999

**Comment by brendankenny** [comment]
It might be good to have known failure cases in the tests to work as TODOs along with the filed issues. e.g. an expectation for `robots.txt` of `"Lighthouse was unable to download a robots.txt file"`. Helps to ensure they're really fixed when they are, but also just a good starting checklist of page features to include in `yarn smoke csp` v1.

> * We should probably deliver the CSP through an HTTP header so we have access to the `sandbox` directive which was causing issues in #11925. Unfortunately this means we need to find a way to deliver the CSP header specifically for this smoke test.

Yet another query parameter (YAQP) would work? OTOH I don't know if we're anywhere close to fixing that issue without outright protocol support, right?

**Comment by adamraine** [comment]
> OTOH I don't know if we're anywhere close to fixing that issue without outright protocol/brower support for more or less disabling a CSP directive for the page, right?

Yeah, but we can still put in the work now to make sure every CSP feature is available for testing.

**Comment by adamraine** [comment]
Didn't know this was available, very cool:
https://github.com/GoogleChrome/lighthouse/blob/e0be31815bbf2c56d8bae8660a4a338e138b80e3/lighthouse-cli/test/fixtures/static-server.js#L124-L132

**Comment by adamraine** [comment]
Keeping open for https://github.com/GoogleChrome/lighthouse/pull/12391#pullrequestreview-647535411

**Comment by brendankenny** [comment]
> Keeping open for [#12391 (review)](https://github.com/GoogleChrome/lighthouse/pull/12391#pullrequestreview-647535411)

Maybe scope out what CSP for dbw would involve before committing to it? There's enough layers of interacting nastiness in dbw that it's difficult to unwind a good bit of it. If adding a decent CSP to dbw would need cutting anything significant from it, I'd personally just be happy with adding a new kitchen-sink smoke test.

**Comment by Seirdy** [comment]
Might be worth testing with an empty `sandbox` CSP directive once https://github.com/GoogleChrome/lighthouse/issues/11925 gets fixed [upstream](https://bugs.chromium.org/p/chromium/issues/detail?id=1222763).
